### ISSUE TITLE
Error executing script. Unexpected error for call_service at pos 1: 'mappingproxy' object does not support item assignment

### ISSUE THREAD

**Issue by Ceralor** [nielsfaber/scheduler-component#82]
Hi there! I wrote an automation to take arguments from a webhook for creating a reminder of sorts, calling `scheduler.add`. However I'm getting an error message, as shown above, trace here:

```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 250, in _async_step
    await getattr(
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 457, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1445, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1484, in _execute_service
    await self._hass.async_add_executor_job(handler.job.target, service_call)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/scheduler/__init__.py", line 77, in service_create_schedule
    coordinator.async_create_schedule(service.data)
  File "/config/custom_components/scheduler/__init__.py", line 208, in async_create_schedule
    res = self.store.async_create_schedule(data)
  File "/config/custom_components/scheduler/store.py", line 189, in async_create_schedule
    data = parse_schedule_data(data)
  File "/config/custom_components/scheduler/store.py", line 79, in parse_schedule_data
    data["timeslots"] = timeslots
TypeError: 'mappingproxy' object does not support item assignment
```

Here's my action calling the service:

```
service: scheduler.add
data:
  weekdays: '{{ (now() + timedelta(minutes=trigger.json.time))| as_timestamp | timestamp_custom("%a") | lower }}'
  repeat_type: single
  timeslots:
    - start: '{{ (now() + timedelta(minutes=trigger.json.time))| as_timestamp | timestamp_custom("%H:%M") }}'
      actions:
        - entity_id: media_player.whole_home
          service: script.metro_announce
          service_data:
            message: '{{ trigger.json.message }}'
```

The requirement for an entity to be specified is a little confusing, because there are numerous services that can be called that do not accept one and may even error from it being specified. I understand this may primarily be due to the way the accompanying card works, and it's quite slick, but it would be nice to allow doing this without requiring the card.

**Comment by nielsfaber** [comment]
I’m guessing here that the error is caused by the fact that you use numerous templates in the service call.
If you use the example data from the service definition, does this work without problems?

Anyway, it’s probably an easy fix.
I guess this is solved by changing this line:
https://github.com/nielsfaber/scheduler-component/blob/fa9541f376697d2373e18c821a3fdd9da6fb6a95/custom_components/scheduler/__init__.py#L77

Try replacing `service.data` by `dict(service.data)` to typecast it (mapping proxy is read only).

I’m not home right now (typing on the phone) and it may take some days before I can release an update.
Hope this helps you.

I could investigate if it’s possible to make the entity_id optional, but I think I need to make some changes here and there for it..

**Comment by Ceralor** [comment]
That dict casting worked for it, thank you!

**Comment by nielsfaber** [comment]
This is fixed in v3.1.0
